src/node_util.cc

```diff
@@ -348,6 +348,69 @@ static void IsInsideNodeModules(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result);
 }
 
+static void DefineLazyPropertiesGetter(
+    Local<v8::Name> name, const v8::PropertyCallbackInfo<Value>& info) {
+  Realm* realm = Realm::GetCurrent(info);
+  Isolate* isolate = realm->isolate();
+  auto context = isolate->GetCurrentContext();
+  Local<Value> arg = info.Data();
+  Local<Value> require_result;
+  if (!realm->builtin_module_require()
```

Comment:
It might be worth to cache the modules in C++ or just pass the `BuiltinModule.map` to C++ land so that multiple exports from the same module do not lead to repeated call backs into the JS `require`.